src/components/abstractComponent.ts

```diff
@@ -502,78 +408,29 @@ export abstract class AbstractComponent<D> implements IComponent<D> {
     this._parent = parent;
     return this;
   }
-  /**
-   * @returns {Bounds} for the component in pixel space, where the topLeft
-   * represents the component's minimum x and y values and the bottomRight represents
-   * the component's maximum x and y values.
-   */
-  public bounds() {
```

Comment:
well if we want behavior to exist on a component then it needs to exist on the interface as that's the definition of the component